src/MonoTorrent.Client/MonoTorrent.Client.Modes/StartingMode.cs

```diff
@@ -156,12 +177,9 @@ async Task TryLoadV2HashesFromCache()
         async Task VerifyHashState ()
         {
             // If we do not have metadata or the torrent needs a hash check, fast exit.
-            if (!Manager.HasMetadata || !Manager.HashChecked)
```

Comment:
We can never reach this point *and* have `Manager.HasMetadata` return false:

https://github.com/alanmcgovern/monotorrent/pull/565/files#diff-ce0c0fad357facaa0ad04c3cdaec3735b709d7ec85caef1e6cf0f5e13183e6aaR61-R64